### PR TITLE
Add approvers & maintainers corner to CONTRIBUTING.md

### PR DESCRIPTION
**Pull request by svrnm**
While chatting with @theletterf, we figured out that it would be good to write down some _practices_ we apply while doing code reviews and PR management. I call them practices because none of this is a must do and there are always exceptions.

@open-telemetry/docs-approvers, all: please take a look!

### CODE REVIEW

File: CONTRIBUTING.md
Comment (theletterf) [inline CONTRIBUTING.md:278]: Any way we can trigger a rebuild using PR comments? What facilities and automations are at our disposal to support contributors?

File: CONTRIBUTING.md
Comment (theletterf) [inline CONTRIBUTING.md:275]: Like two weeks? Can we add a sample period?

File: CONTRIBUTING.md
Comment (theletterf) [inline CONTRIBUTING.md:279]: In the unfortunate event that something get merged and shouldn't, what do we usually do?

File: CONTRIBUTING.md
Comment (theletterf) [inline CONTRIBUTING.md:277]: Any CLA related tip or troubleshooting? What if contributors can't figure it out or are stuck?

**Review by theletterf — APPROVED** [review]
Thanks —this is very helpful! I wager it'll grow over time.

File: CONTRIBUTING.md
Comment (cartermp) [reply CONTRIBUTING.md:275]: Yes, let's go with two weeks.

File: CONTRIBUTING.md
Comment (cartermp) [reply CONTRIBUTING.md:277]: I think for this they need to enable maintainer permissions to update their branch?

File: CONTRIBUTING.md
Comment (svrnm) [reply CONTRIBUTING.md:275]: I kept it out on purpose to leave this to our judgement, but I can say something like "2 weeks in general", allowing us to wait longer (probably never the problem), but also merge somthing urgent quicker without getting in discussions

File: CONTRIBUTING.md
Comment (svrnm) [reply CONTRIBUTING.md:277]: Yeah, CLA is a thing on it's own. If they did a mistake with their email address in one commit to their PR (most common issue), we can either ask them to fix this themselves or rebase the PR (which requires write permissions as @cartermp said). Sometimes CLA is just behaving strange/unexpected, in the worst case the PR needs to be closed & re-opened to re-trigger the checks

File: CONTRIBUTING.md
Comment (theletterf) [inline CONTRIBUTING.md:275]:
```suggestion
  * If no SIG approval is given within a two-week grace period, docs maintainer may use their own judgement to merge that PR
```

File: CONTRIBUTING.md
Comment (theletterf) [inline CONTRIBUTING.md:277]:
```suggestion
* A PR by non-maintainers should **never** update git sub modules. This happens by accident from time to time. Let the PR author know that they should not worry about it, we will fix this before merging, but in the future they should make sure that they work from an up-to-date fork.
* If the contributor is having trouble signing the CLA or used the wrong email by mistake in one of their commits, ask them to fix the issue or rebase the pull request. Worst case scenario, close and re-open the PR to trigger a new CLA check.
```

File: CONTRIBUTING.md
Comment (svrnm) [reply CONTRIBUTING.md:278]: A new commit triggers a rebuild, that's all we have, additionally there is /fix: with refcache and format. Let's have this in a follow up PR

**Comment by svrnm** [comment]
Please take another look

I will create a follow up PR for open questions:
* https://github.com/open-telemetry/opentelemetry.io/pull/3632#discussion_r1412343877
* https://github.com/open-telemetry/opentelemetry.io/pull/3632#discussion_r1412344685

**Review by cartermp — APPROVED** [review]

**Pull request merged**
